code/iaas/lb-logic/src/main/java/io/cattle/iaas/lb/process/LoadBalancerRemovePostListener.java

```diff
@@ -0,0 +1,71 @@
+package io.cattle.iaas.lb.process;
+
+import io.cattle.platform.core.constants.LoadBalancerConstants;
+import io.cattle.platform.core.dao.GenericMapDao;
+import io.cattle.platform.core.model.LoadBalancer;
+import io.cattle.platform.core.model.LoadBalancerConfig;
+import io.cattle.platform.core.model.LoadBalancerConfigListenerMap;
+import io.cattle.platform.core.model.LoadBalancerListener;
+import io.cattle.platform.engine.handler.HandlerResult;
+import io.cattle.platform.engine.handler.ProcessPostListener;
+import io.cattle.platform.engine.process.ProcessInstance;
+import io.cattle.platform.engine.process.ProcessState;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.process.common.handler.AbstractObjectProcessLogic;
+import io.cattle.platform.util.type.Priority;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+@Named
+public class LoadBalancerRemovePostListener extends AbstractObjectProcessLogic implements ProcessPostListener, Priority {
+    @Inject
+    GenericMapDao mapDao;
+
+    @Inject
+    ObjectManager objectManager;
+
+    @Override
+    public String[] getProcessNames() {
+        return new String[] { LoadBalancerConstants.PROCESS_LB_REMOVE };
+    }
+
+    @Override
+    public HandlerResult handle(ProcessState state, ProcessInstance process) {
+        LoadBalancer lb = (LoadBalancer) state.getResource();
+        if (lb.getServiceId() == null) {
```

Comment:
No. This handler takes care only of Service's load balancer. I should probably change the name of the handler to reflect that.